lib/mix/test/mix/tasks/compile.app_test.exs

```diff
@@ -97,19 +130,19 @@ defmodule Mix.Tasks.Compile.AppTest do
       end
 
       Process.put(:application, [applications: ["invalid"]])
-      message = "Application dependencies (:applications) should be a list of atoms, got: [\"invalid\"]"
+      message = "Application applications (:applications) should be a list of atoms, got: [\"invalid\"]"
       assert_raise Mix.Error, message, fn ->
         Mix.Tasks.Compile.App.run([])
       end
 
       Process.put(:application, [applications: nil])
-      message = "Application dependencies (:applications) should be a list of atoms, got: nil"
+      message = "Application applications (:applications) should be a list of atoms, got: nil"
       assert_raise Mix.Error, message, fn ->
         Mix.Tasks.Compile.App.run([])
       end
 
       Process.put(:application, [env: [:invalid]])
-      message = "Application dependencies (:env) should be a keyword list, got: [:invalid]"
+      message = "Application environment (:env) should be a keyword list, got: [:invalid]"
       assert_raise Mix.Error, message, fn ->
         Mix.Tasks.Compile.App.run([])
       end
```

Comment:
Should we add a test here for validation of `:extra_applications` as well? We have it for all other options, it may be weird to skip this one. 😕